mafia-react/src/pages/GameRoom.tsx

```diff
@@ -133,61 +136,68 @@ export const GameRoom = (props: Props) => {
           <Grid container className={classes.root} spacing={2}>
             {data?.data.players.map((player: PlayerType, index: number) => {
               const isUser = player.userId === Cookies.get('userId');
-              return (
-                <Grid
-                  item
-                  key={index}
-                  data-is-user={isUser}
-                  style={{ order: isUser ? -1 : 0 }}
-                >
-                  <PlayerCard
-                    name={player.name}
-                    role={
-                      player.userId === Cookies.get('userId') ||
-                      data?.data.roomMaster === Cookies.get('userId') ||
-                      (playerData !== undefined &&
-                        playerData.role === 'mafia' &&
-                        player.role === 'mafia') ||
-                      data?.data.status === 'ended'
-                        ? player.role
-                        : '???'
-                    }
-                    trueRole={player.role}
-                    checked={
-                      playerData !== undefined && playerData.role === 'detective'
-                        ? player.checked
-                        : false
-                    }
-                    status={player.status}
-                    isUser={player.userId === Cookies.get('userId')}
-                    phase={data?.data.phase}
-                    onKill={async (event) =>
-                      await killRequest(params.roomId, player.userId).catch((err) => {
-                        if (err.response.status >= 400)
-                          showErrorMessage(err.response.data.message);
-                      })
-                    }
-                    onHeal={async (event) =>
-                      await healRequest(params.roomId, player.userId).catch((err) => {
-                        if (err.response.status >= 400)
-                          showErrorMessage(err.response.data.message);
-                      })
-                    }
-                    onCheck={async (event) =>
-                      await checkRequest(params.roomId, player.userId).catch((err) => {
-                        if (err.response.status >= 400)
-                          showErrorMessage(err.response.data.message);
-                      })
-                    }
-                    onHang={async (event) =>
-                      await voteRequest(params.roomId, player.userId).catch((err) => {
-                        if (err.response.status >= 400)
-                          showErrorMessage(err.response.data.message);
-                      })
-                    }
-                  />
-                </Grid>
-              );
+              if (
+                ((Date.now() - player.last_poll.$date) / 1000 < 4 &&
+                  data?.data.phase === 'pre-game') ||
+                data?.data.phase !== 'pre-game'
+              ) {
```

Comment:
Having trouble understanding the logic here. So always show the grid of players, except for if you're inactive during an ongoing game? :
```suggestion
const isPlayerActive = (Date.now() - player.last_poll.$date) / 1000 < 4;
const startedGameAlready = data?.data.phase === 'pre-game';
const awaitingGameStart = data?.data.phase !== 'pre-game';
              if ((isPlayerActive && startedGameAlready) || awaitingGameStart) {
```

Reply:
so basically, if it is pre-game, we want the app to be able to see how many players to include in the game, rather than people just joining and decide to leave last minute, and then people have to make a new room. So if it is pre-game, people who join will always be added to the list, but we need last_poll to see if they are active. So if they close the tab or something the app will detect that a player is not active. So that when you press 'Start Game' the app only takes in those players who are online, and removes everyone who is offline.

Reply:
if the game is in progress, we don't care who is active or not anymore because they are already in the game, so we want to show everyone regardless of whether someone is online or has closed the tab accidentally and wants to come back or whatnot. We still want to show everyone when game is in-progress.

Reply:
ah i see, i had the variables named backwards. 
so `if ((isPlayerActive && awaitingGameStart) || startedGameAlready) {`